src/main/java/org/folio/inventory/dataimport/consumers/QuickMarcKafkaHandler.java

```diff
@@ -80,21 +88,57 @@ private void createProducer(KafkaConfig kafkaConfig, QMEventTypes eventType) {
       LOGGER.info(format("Quick marc event payload has been received with event type: %s", event.getEventType()));
       kafkaInternalCache.putToCache(event.getId());
       return getEventPayload(event)
-        .compose(eventPayload -> new UpdateInstanceEventHandler(instanceUpdateDelegate, context, precedingSucceedingTitlesHelper).handle(eventPayload)
-            .compose(ar -> sendEvent(eventPayload, QM_INVENTORY_INSTANCE_UPDATED, params))
-            .recover(throwable -> {
-              eventPayload.put("ERROR", throwable.getMessage());
-              return sendEvent(eventPayload, QM_ERROR, params);
-            })
-            .map(ar -> record.key()), th -> Future.failedFuture(th.getMessage()
-          )
+        .compose(eventPayload -> processPayload(eventPayload, context)
+          .compose(recordType -> sendEvent(eventPayload, getReplyEventType(recordType), params))
+          .recover(throwable -> sendErrorEvent(params, eventPayload, throwable))
+          .map(ar -> record.key()),
+          th -> Future.failedFuture(th.getMessage())
         );
     }
     return Future.succeededFuture();
   }
 
+  private QMEventTypes getReplyEventType(Record.RecordType recordType) {
+    if (recordType == Record.RecordType.MARC_BIB) {
```

Comment:
If we have` Record.RecordType.MARC_AUTHORITY` or `Record.RecordType.EDIFACT`, will be work normal for these types?
Maybe we should use `else if`

Reply:
quick marc doesn't work with authority or edifact records.